src/handlers/CrossMarginHandler02.sol

```diff
@@ -0,0 +1,606 @@
+
+    // Sanity check
+    CrossMarginService(_crossMarginService).vaultStorage();
+    IEcoPyth(_pyth).getAssetIds();
+
+    crossMarginService = _crossMarginService;
+    pyth = _pyth;
+    minExecutionOrderFee = _minExecutionOrderFee;
+  }
+
+  /**
+   * Modifiers
+   */
+
+  /// @notice Validate only accepted collateral tokens to be deposited or withdrawn
+  modifier onlyAcceptedToken(address _token) {
+    ConfigStorage(CrossMarginService(crossMarginService).configStorage()).validateAcceptedCollateral(_token);
+    _;
+  }
+
+  /// @notice Validate only whitelisted executors to call function
+  modifier onlyOrderExecutor() {
+    if (!orderExecutors[msg.sender]) revert ICrossMarginHandler02_NotWhitelisted();
+    _;
+  }
+
+  modifier delegate(address _mainAccount) {
+    if (delegations[_mainAccount] == msg.sender) {
+      _senderOverride = _mainAccount;
+    }
+    _;
+    _senderOverride = address(0);
+  }
+
+  function _msgSender() internal view override returns (address) {
+    if (_senderOverride == address(0)) {
+      return msg.sender;
+    } else {
+      return _senderOverride;
+    }
+  }
+
+  /**
+   * Deposit Collateral
+   */
+
+  /// @notice Deposits the specified amount of collateral token into the user's sub-account.
+  /// @param _subAccountId ID of the user's sub-account.
+  /// @param _token Address of the collateral token to deposit.
+  /// @param _amount Amount of collateral token to deposit.
+  /// @param _shouldWrap Whether to wrap native ETH into WETH before depositing.
+  function depositCollateral(
+    address _mainAccount,
+    uint8 _subAccountId,
+    address _token,
+    uint256 _amount,
+    bool _shouldWrap
+  ) external payable nonReentrant onlyAcceptedToken(_token) delegate(_mainAccount) {
+    if (_amount == 0) revert ICrossMarginHandler02_BadAmount();
+    // SLOAD
+    CrossMarginService _crossMarginService = CrossMarginService(crossMarginService);
+
+    if (_shouldWrap) {
+      // Prevent mismatch msgValue and the input amount
+      if (msg.value != _amount) {
+        revert ICrossMarginHandler02_MismatchMsgValue();
+      }
+
+      // Wrap the native to wNative. The _token must be wNative.
+      // If not, it would revert transfer amount exceed on the next line.
+      // slither-disable-next-line arbitrary-send-eth
+      IWNative(_token).deposit{ value: _amount }();
+      // Transfer those wNative token from this contract to VaultStorage
+      ERC20Upgradeable(_token).safeTransfer(_crossMarginService.vaultStorage(), _amount);
+    } else {
+      // Transfer depositing token from trader's wallet to VaultStorage
+      ERC20Upgradeable(_token).safeTransferFrom(_msgSender(), _crossMarginService.vaultStorage(), _amount);
+    }
+
+    // Call service to deposit collateral
+    _crossMarginService.depositCollateral(_msgSender(), _subAccountId, _token, _amount);
+
+    emit LogDepositCollateral(_msgSender(), _subAccountId, _token, _amount);
+  }
```

Comment:
In the new `Ext01Handler` and the existing `LimitTradeHandler`, there is also this check:

```solidity
if (_mainAccount != _msgSender()) revert();
``` 

Would it make sense to add this to the `CrossMarginHandler02` functions too? It's not necessarily a bug to exclude it, (because `_mainAccount` is only used in the `delegate()` modifier here), but maybe it was intended to be here